src/scheduler.rs

```diff
@@ -89,21 +99,41 @@ impl ScheduledPayJoin {
             .iter()
             .map(|channel| channel.amount)
             .fold(bitcoin::Amount::ZERO, std::ops::Add::add);
-        // TODO: replace with sheduled_payjoin.fees()
-        let fees = calculate_fees(
-            self.channels.len() as u64,
-            self.fee_rate,
-            self.wallet_amount() != bitcoin::Amount::ZERO,
-        );
-        let wallet_amount = self.wallet_amount();
+        let reserve_deposit = self.reserve_deposit();
+        let quote_amount = match &self.quote {
+            Some(quote) => Amount::from_sat(quote.price.into()),
+            None => Amount::ZERO,
+        };
+
+        (total_channel_amount + reserve_deposit + quote_amount + self.fees()).as_sat()
+            == our_output.value
+    }
 
-        let owned_txout_value = our_output.value;
+    /// This externally exposes [ScheduledPayJoin]::reserve_deposit.
+    pub fn reserve_deposit(&self) -> bitcoin::Amount { self.reserve_deposit }
 
-        (total_channel_amount + fees + wallet_amount).as_sat() == owned_txout_value
-    }
+    /// Calculate the absolute miner fee this [ScheduledPayJoin] pays
+    fn fees(&self) -> bitcoin::Amount {
+        let channel_count = self.channels.len() as u64;
+        let has_reserve_deposit = self.reserve_deposit != bitcoin::Amount::ZERO;
 
-    /// This externally exposes [ScheduledPayJoin]::wallet_amount.
-    pub fn wallet_amount(&self) -> bitcoin::Amount { self.wallet_amount }
+        let mut additional_vsize = if has_reserve_deposit {
+            // <8 invariant bytes = 4 version + 4 locktime>
+            //  + 2 variant bytes for input.len + output.len such that each len < 252
+            //  + OP_0 OP_PUSHBYTES_32 <32 byte script>
+            channel_count * (8 + 1 + 1 + 34)
+        } else {
+            // substitute 1 p2wsh channel (34 bytes) open for 1 p2wpkh reserve output (22 bytes)
+            // that's + 12 bytes
+            (channel_count - 1) * (8 + 1 + 1 + 34) + 12
+        };
+
+        if self.quote.is_some() {
+            additional_vsize = additional_vsize + (8 + 1 + 1 + 22); // P2WPKH (OP_0 OP_PUSHBYTES_20 <20 byte script)
+        }
+
+        bitcoin::Amount::from_sat(self.fee_rate * additional_vsize)
```

Comment:
slightly excessive fee calculation but OK for this PR. The sender approves the fee anyhow